ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py

```diff
@@ -214,93 +189,39 @@ def yield_dashboard_lineage_details(
                 logger.debug(traceback.format_exc())
                 logger.error(f"Error creating chart [{chart}]: {exc}")
 
-    def req_get(self, path):
-        """Send get request method
-
-        Args:
-            path:
-        """
-        return requests.get(
-            self.service_connection.hostPort + path,
-            headers=self.metabase_session,
-            timeout=30,
-        )
-
     def _yield_lineage_from_query(
         self, chart_details: dict, db_service_name: str, dashboard_name: str
     ) -> Optional[AddLineageRequest]:
-        resp_database = self.req_get(f"/api/database/{chart_details['database_id']}")
-        if resp_database.status_code == 200:
```

Comment:
Note: change in logic: I have removed the is statements which checks the status code, from here,

In order to move the `get` logic to client.py, I moved the database.json() function to client.py.
Unintended effect: I miss one if statement which checks status code.

Any suggestions?

Idea 1: move the if statement logic to client.py and return None in the else part
Idea 2: wrap it with a try-except block.

Question: why was there an if statement in the first place?

Reply:
the client should be the one validating the API response indeed. Returning an `Optional` IMO is the right way, and we can handle None in the ingestion